### PR TITLE
Searchmdn: Add command to search MDN

### PR DESCRIPTION
**Pull request by Asartea**
## Because
- When helping users, we often encourage them to look at an outside resource, which most of the time turns out to be MDN. While this functionality can already be achieved by using `!google mdn <search term>`, its an common enough occurrence that its worth making it its own command, both for ease of use and because it lets us use MDN's own search instead of relying on google

## This PR
- Adds a new old style `!mdn` command, as well as a `/searchmdn` command
 - I opted to also add an old style command because I expect the primary usage of this will be within text flow, with something like "You might also want to look into using `.split()`; !mdn Array.split" 

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Callbacks command: Update verbiage`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR

### CODE REVIEW

**Comment by Arkanheal** [comment]
Are you sure we want to answer to be embed for the "legacy" command (I don't mind for the slash one)? It kinda breaks the flow of the conversation you were talking about.

File: new-era-commands/slash/searchmdn.js
Comment (Arkanheal) [inline new-era-commands/slash/searchmdn.js:23]: I don't think we want to "reply" to the user that entered the command most of the time? Or maybe it could be a use-case specific to the slash command, a way to get a link?

**Comment by Asartea** [comment]
See my thinking here was twofold: A) @ManonLef has already been doing some work on converting other legacy ones over, so otherwise this one will at some point stand out as weird, and B) honestly I think an embed like this is less jarring than a random message by odin-bot

**Comment by Arkanheal** [comment]
> See my thinking here was twofold: A) @ManonLef has already been doing some work on converting other legacy ones over, so otherwise this one will at some point stand out as weird, and B) honestly I think an embed like this is less jarring than a random message by odin-bot

But if we embed anyway and convert legacy commands to new-era ones, why not just create only a slash from the start?

**Comment by Asartea** [comment]
> But if we embed anyway and convert legacy commands to new-era ones, why not just create only a slash from the start?

Because if I am helping an user I still want to be able to enter `!mdn something`, rather than having to send a message, and then send a slash command. Willing to be convinced on this point however

File: new-era-commands/slash/searchmdn.js
Comment (Asartea) [reply new-era-commands/slash/searchmdn.js:23]: This is the same behavior as with all other slash commands:

image

Comment (Arkanheal) [reply new-era-commands/slash/searchmdn.js:23]: Oh yeah true 👍

File: botCommands/searchmdn.js
Comment (Arkanheal) [inline botCommands/searchmdn.js:4]: Adding a comment here, so we can keep the PR feed clean.

Comment (Arkanheal) [reply botCommands/searchmdn.js:4]: TBH I think having both at the same time could be confusing (specially to new helpers). Given that we are apparently rewriting everything to ~~Rust~~ slash embed commands, I'd rather have only one. @Asartea

Comment (Arkanheal) [reply botCommands/searchmdn.js:4]: For completeness, I have the same remarks about the `!google` [PR](https://github.com/TheOdinProject/odin-bot-v2/pull/388)

Comment (Asartea) [reply botCommands/searchmdn.js:4]: Yeah preserving some of them as `!` commands is already the plan (see #297 ). I think this should be one of them, but thats of course just my opinion. @01zulfi do you have any thoughts on this?

**Comment by 01zulfi** [comment]
@Asartea sorry for taking so long.

We don't need a text based command for this. A slash command would do fine. Let's gather some of the team's thoughts before merging.

@TheOdinProject/moderators @TheOdinProject/maintainers what do we think about a `/searchmdn` command?

**Comment by JustWaveThings** [comment]
I'd love a /searchmdn command

File: botCommands/searchmdn.js
Comment (01zulfi) [inline botCommands/searchmdn.js:1]: @Asartea Looks like we're gonna add the command.

Could you please remove the text-based command as the slash command gets the job done.

**Review by 01zulfi — CHANGES_REQUESTED** [review]

File: botCommands/searchmdn.js
Comment (Asartea) [reply botCommands/searchmdn.js:1]: Done

**Review by 01zulfi — APPROVED** [review]
lgtm 🚀

**Pull request merged**
